README.md

```diff
@@ -16,6 +16,14 @@ See [Choosing Secure Passwords](https://www.schneier.com/blog/archives/2014/03/c
 This feature enables the HTTP header `X-XSS-Protection: 1; mode=block` to be sent on all requests which some web browsers intend as a way to automatically block suspected cross-site scripting attacks.
 Several web browsers (e.g., Firefox, Edge, and Chrome) do not support this header.
 
+## Remove HTTP Headers for Unauthorized Users
+
+This feature adds a configurable list of HTTP header names (case-insensitive) that can be automatically removed from HTTP responses sent to users lacking the *Overall/Read* permission.
+This is useful for hiding the Jenkins and other software versions present in HTTP headers like `X-Jenkins` and `Server`.
+Various security scanning tools recommend hiding version information advertised this way.
+Note there are numerous other ways to fingerprint Jenkins to deduce which version is running even when this feature is enabled.
+See also the [Jetty header filter](https://www.eclipse.org/jetty/documentation/current/header-filter.html) for a more advanced set of configurable header filters.
```

Comment:
It might also be useful to mention using a proxy or whatever to accomplish the same thing. I'm not sure how to word it.